corehq/messaging/scheduling/models/timed_schedule.py

```diff
@@ -150,11 +188,13 @@ def get_local_next_event_due_timestamp_for_monthly_schedule(self, instance):
                 # always schedule a negative day.
                 raise InvalidMonthlyScheduleConfiguration("Day must be between -28 and 31, and not be 0")
 
-            year_offset = (instance.schedule_iteration_num - 1) // 12
-            month_offset = (instance.schedule_iteration_num - 1) % 12
+            months_since_start_date = (instance.schedule_iteration_num - 1) * (-1 * self.repeat_every)
```

Comment:
nit: could simply negate the variable (`-self.repeat_every`) rather than multiplying by `-1` (same applies in several other places as well)

Reply:
yeah i've never been too crazy about that syntax, mainly because it opens up the risk of someone changing this line later and accidentally turning it into a subtraction